src/vos/vos_obj.c

```diff
@@ -2029,10 +2142,32 @@ vos_obj_iter_aggregate(daos_handle_t ih, bool range_discard)
 
 		rc = dbtree_iter_delete(oiter->it_hdl, NULL);
 		D_ASSERT(rc != -DER_NONEXIST);
-	} else if (rc == -DER_NONEXIST) {
-		/* Key no longer exists at epoch but isn't empty */
-		invisible = true;
-		rc = 0;
+	} else {
+		if (rc == -DER_NONEXIST) {
+			/* Key no longer exists at epoch but isn't empty */
+			invisible = true;
+			rc = 0;
+		}
+		if (krec->kr_bmap & KREC_BF_BTR)
+			feats = dbtree_feats_get(&krec->kr_btr);
+		else
+			feats = evt_feats_get(&krec->kr_evt);
+		if (oiter->it_iter.it_for_purge && feats & VOS_TREE_AGG_FLAG) {
+			/** We got through aggregation without anyone clearing the flag,
+			 *  so we can clear the needed flag if we scanned the whole subtree.
+			 *  Either way, we can clear the aggregation flag.
+			 */
+			feats = feats & ~VOS_TREE_AGG_FLAG;
+			if (!oiter->it_iter.it_skipped)
```

Comment:
I suppose if we use the upper bits of the timestamp, we don't have to worry about wrap.  And we can even use more than 32 because we have about 60 feature bits to spare.   Perhaps we could just use the physical clock component of the HLC or upper 46 bits and store it in the same place I'm storing this flag.   And it could retain the aggregatable meaning so rather than last write, it's last aggregatable write